### PR TITLE
Add data path annotation section to ARC specification

### PR DESCRIPTION
**Pull request by HLWeil**
#56 
#59 

Implemented as often discussed. Feedback would be very welcome, as this is probably also one of the most frequently asked questions

@kappe-c @kMutagene @muehlhaus @chgarth @Brilator

### CODE REVIEW

**Comment by Brilator** [comment]
Looks like a good solution to me (especially with the backdoor for dataset / resources).

File: ARC specification.md [hunk 1]
```diff
@@ -232,6 +233,37 @@ The file `arc.cwl` MUST exist at the root directory of each ARC. It describes wh
 
 `arc.cwl` MUST be a CWL v1.2 workflow description and adhere to the same requirements as [run descriptions](#run-description). In particular, references to study or assay data files, nested workflows MUST use relative paths. An optional file `arc.yml` MAY be provided to specify input parameters.
 
+### Data Path Annotation
+
+All metadata references to files or directories located inside the ARC MUST follow the following patterns:
+
+- The `general pattern`, which is always applicable and SHOULD always be used is to specify the path relative to the ARC root
```
Comment (Brilator) [inline ARC specification.md:240]: Maybe rephrase the "which is always applicable". 
Even though the SHOULD is clearly not a MUST, to me the "always" sounds like this must always be done, which would contradict below folder-specific pattern.

Comment (HLWeil) [reply ARC specification.md]: Thanks for you input! 
IMO the `SHOULD` is really a SHOULD as the `general pattern` is way more verbose so it probably came off exactly like planned. But I still softened it up a bit. Does this match more what you would expect now?

PS: Ho Ho Ho! 🎅

File: ARC specification.md [hunk 2]
```diff
@@ -232,6 +233,37 @@ The file `arc.cwl` MUST exist at the root directory of each ARC. It describes wh
 
 `arc.cwl` MUST be a CWL v1.2 workflow description and adhere to the same requirements as [run descriptions](#run-description). In particular, references to study or assay data files, nested workflows MUST use relative paths. An optional file `arc.yml` MAY be provided to specify input parameters.
 
+### Data Path Annotation
+
+All metadata references to files or directories located inside the ARC MUST follow the following patterns:
+
+- The `general pattern`, which is universally applicable and SHOULD be used is to specify the path relative to the ARC root
+
+- The `folder specific pattern`, which MAY be used. This pattern dependes on the metadata context:
```
Comment (kMutagene) [inline ARC specification.md:242]: which MAY be used <in the following circumstances> ?

Comment (kMutagene) [reply ARC specification.md:242]: notsure about the wording but i think the sentence currently could imply that it MAY be used all the time, even if you specify cases afterwards

Comment (HLWeil) [reply ARC specification.md]: I collapsed the two sentences into one, hopefully explicating the constraint
